runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java

```diff
@@ -5149,13 +5149,15 @@ private int onDecodeData(
                         if (!HttpState.replyOpened(exchange.state))
                         {
                             doEncodeHeaders(traceId, authorization, streamId, headers400, true);
+                            doNetworkEnd(traceId, authorization);
```

Comment:
Calling `doNetworkEnd` will close the underlying `tcp` connection, which is heavy handed because there can be other HTTP/2 requests in flight over the same connection.

The HTTP/2 equivalent of this is `doEncodeRstStream`, so it seems that we need to call that here as well.

In fact, I think we just need to make the `doEncodeRstStream` call below unconditional and remove the call to `doNetworkEnd` here.